Library/Homebrew/keg.rb

```diff
@@ -295,14 +295,24 @@ def lock
   end
 
   def completion_installed?(shell)
-    dir = case shell
-    when :bash then path.join("etc", "bash_completion.d")
-    when :zsh  then path.join("share", "zsh", "site-functions")
-    when :fish then path.join("share", "fish", "vendor_completions.d")
+    case shell
+    when :bash
+      dir = path.join("etc", "bash_completion.d")
+    when :zsh
+      # By convention, zsh completion files start with an underscore
+      return !Dir["#{path}/share/zsh/site-functions/_*"].empty?
```

Comment:
This early return inside a `case` statement is a bit weird. Could you rework it so it shares the logic below?